src/robusta/core/sinks/sink_base.py

```diff
@@ -33,20 +34,63 @@ def register_notification(self, interval: int, threshold: int) -> bool:
 
 class NotificationSummary(BaseModel):
     message_id: Optional[str] = None  # identifier of the summary message
-    start_ts: float = Field(default_factory=lambda: time.time())  # Timestamp of the first notification
+    start_ts: float = None
+    end_ts: float = None
     # Keys for the table are determined by grouping.notification_mode.summary.by
     summary_table: DefaultDict[KeyT, List[int]] = None
 
-    def register_notification(self, summary_key: KeyT, resolved: bool, interval: int):
-        now_ts = time.time()
+    def register_notification(self, summary_key: KeyT, resolved: bool, interval: int, aligned: bool):
+        now_dt = datetime.now()
+        now_ts = int(now_dt.timestamp())
         idx = 1 if resolved else 0
-        if now_ts - self.start_ts > interval or not self.summary_table:
-            # Expired or the first summary ever for this group_key, reset the data
+        if not self.end_ts or now_ts > self.end_ts:
```

Comment:
what if not self.summary_table?

Reply:
If `end_ts` is not set it means that the whole object is not initialized, and so `summary_table` is also not set. I basically simplified the condition here.